### PR TITLE
KEYCLOAK-5186 createUser: set federationLink

### PR DESCRIPTION
**Pull request by wvdhaute**

### CODE REVIEW

**Comment by stianst** [comment]
Please add tests

**Comment by stianst** [comment]
This is not how we tests things in Keycloak. Please write proper functional tests instead of mocks.

**Comment by stianst** [comment]
https://github.com/keycloak/keycloak/blob/master/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/UserTest.java

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/AbstractAdminTest.java [hunk 1]
```diff
@@ -71,6 +75,16 @@ public void addTestRealms(List<RealmRepresentation> testRealms) {
         config.put("port", "3025");
         adminRealmRep.setSmtpServer(config);
 
+        // add dummy federation provider
```
Comment (stianst) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/AbstractAdminTest.java]: Don't add a dummy federation provider here as it affects a large amount of other tests. Use the admin client in the one test you need it in to add it

**Review by stianst — CHANGES_REQUESTED** [review]

**Review by stianst — CHANGES_REQUESTED** [review]
DummyUserFederationProviderFactory is already part of testsuite-providers which is added to the server. There's no need to add additional service definition file. Also don't add the provider in AbstractAdminTest as that will affect many tests that don't need it. Use the adminClient to add it where needed.

**Comment by stianst** [comment]
@wvdhaute are you able to do the requested changes?

**Comment by wvdhaute** [comment]
@stianst sorry, had some time off.
Just rebased and squashed an update for the test. Should be ok now

**Review by stianst — APPROVED** [review]

**Comment by stianst** [comment]
Thanks

**Pull request merged**